content-based-message-routing/README.md

```diff
@@ -0,0 +1,497 @@
+                        hospital_id = hospital_id,
+                        patient = patient.name,
+                        doctor = reservationRequest.doctor);
+
+        http:Client hospitalEP;
+        match hospital_id {
+            GRAND_OAK => {
+                hospitalEP = grandOakEP;
+            }
+            CLEMENCY => {
+                hospitalEP = clemencyEP;
+            }
+            _ => {
+                hospitalEP = pineValleyEP;
+            }
+        }
+
+        ReservationResponse|http:ClientError resp = hospitalEP->/[category]/reserve.post({
+            patient,
+            ...reservationRequest
+        });
+
+        if resp is ReservationResponse {
+            log:printDebug("Reservation request successful",
+                            name = patient.name,
+                            appointmentNumber = resp.appointmentNumber);
+            return resp;
+        }
+
+        log:printError("Reservation request failed", resp);
+        if resp is http:ClientRequestError {
+            return <http:NotFound> {body: "Unknown hospital, doctor or category"};
+        }
+
+        return <http:InternalServerError> {body: resp.message()};
+    }
+}
+```
+
+- Define the `hospitalEP` variable. Later, the relevant client is assigned to this variable based on the `hospital_id` value.
+
+    ```ballerina
+    http:Client hospitalEP;
+    ```
+
+- Use a [typed binding pattern with a mapping binding pattern to destructure](https://ballerina.io/learn/by-example/rest-binding-pattern-in-mapping-binding-pattern/) the payload and assign required components of the value to separate variables.
+    
+    ```ballerina
+    ReservationRequest {hospital_id, patient, ...reservationRequest} = payload;
+    ```
+    
+    Here,
+    - The `hospital_id` value is assigned to the `hospital_id` variable. 
+    - The `patient` value is assigned to the `patient` variable of type `Patient` (fields are exactly those expected by the `Patient` record). 
+    - The remaining components of the payload are collected in a mapping value and assigned to the `reservationRequest` variable.
+
+- The `log` functions are used to [log](https://ballerina.io/learn/by-example/#log) information at `INFO`, `DEBUG`, and `ERROR` log levels.
+
+- Use a [match statement](https://ballerina.io/learn/by-example/match-statement) to assign the relevant client to the `hospitalEP` variable based on the `hospital_id` value.
+
+    ```ballerina
+        match hospital_id {
+            GRAND_OAK => {
+                hospitalEP = grandOakEP;
+            }
+            CLEMENCY => {
+                hospitalEP = clemencyEP;
+            }
+            _ => {
+                hospitalEP = pineValleyEP;
+            }
+        }
+    ```
+
+- The backend call is a `POST` request to the backend hospital service to reserve the appointment. The `category` value is used as a path parameter. The response is assigned to the `resp` variable.
```

Comment:
```suggestion
- Make a `POST` request to the backend hospital service to reserve the appointment. The `category` value is used as a path parameter. The response is assigned to the `resp` variable.
```

Reply:
Need to describe the payload/request also?

Reply:
Updated via https://github.com/ballerina-guides/integration-tutorials/pull/4/commits/6834764e3b43b15eefecde1abe244750d8056bc9. Please check.